src/adapters/vertexai/chat.ts

```diff
@@ -0,0 +1,208 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AsyncStream,
+  BaseLLMTokenizeOutput,
+  ExecutionOptions,
+  GenerateCallbacks,
+  GenerateOptions,
+  LLMCache,
+  LLMMeta,
+} from "@/llms/base.js";
+import { shallowCopy } from "@/serializer/utils.js";
+import type { GetRunContext } from "@/context.js";
+import { Emitter } from "@/emitter/emitter.js";
+import { VertexAI, GenerativeModel } from "@google-cloud/vertexai";
+import { ChatLLM, ChatLLMOutput } from "@/llms/chat.js";
+import { BaseMessage, Role } from "@/llms/primitives/message.js";
+import { Serializer } from "@/serializer/serializer.js";
+import { getPropStrict } from "@/internals/helpers/object.js";
+import { signalRace } from "@/internals/helpers/promise.js";
+
+export class VertexAIChatLLMOutput extends ChatLLMOutput {
+  public readonly chunks: BaseMessage[] = [];
+
+  constructor(chunk: BaseMessage) {
+    super();
+    this.chunks.push(chunk);
+  }
+
+  get messages() {
+    return this.chunks;
+  }
+
+  merge(other: VertexAIChatLLMOutput): void {
+    this.chunks.push(...other.chunks);
+  }
+
+  getTextContent(): string {
+    return this.chunks.map((result) => result.text).join("");
+  }
+
+  toString(): string {
+    return this.getTextContent();
+  }
+
+  createSnapshot() {
+    return { chunks: shallowCopy(this.chunks) };
+  }
+
+  loadSnapshot(snapshot: ReturnType<typeof this.createSnapshot>): void {
+    Object.assign(this, snapshot);
+  }
+}
+
+// Feel free to extend if you want to support additional parameters
+type VertexAIGenerateOptions = GenerateOptions;
+
+export interface VertexAIChatLLMInput {
+  modelId: string;
+  project: string;
+  location: string;
+  executionOptions?: ExecutionOptions;
+  cache?: LLMCache<VertexAIChatLLMOutput>;
+  parameters?: Record<string, any>;
+}
+
+export class VertexAIChatLLM extends ChatLLM<VertexAIChatLLMOutput> {
+  public readonly emitter: Emitter<GenerateCallbacks> = Emitter.root.child({
+    namespace: ["vertexai", "llm"],
+    creator: this,
+  });
+
+  protected client: GenerativeModel;
+
+  constructor(protected readonly input: VertexAIChatLLMInput) {
+    super(input.modelId, input.executionOptions, input.cache);
+    const vertexAI = new VertexAI({ project: input.project, location: input.location });
+    this.client = vertexAI.getGenerativeModel({
+      model: input.modelId,
+    });
+  }
+
+  static {
+    this.register();
+    Serializer.register(GenerativeModel, {
+      toPlain: (value) => ({
+        project: getPropStrict(value, "project"),
+        location: getPropStrict(value, "location"),
+        model: getPropStrict(value, "model"),
+        safetySettings: getPropStrict(value, "safetySettings"),
+        generationConfig: getPropStrict(value, "generationConfig"),
+        // systemInstruction: getPropStrict(value, "systemInstruction"),
```

Comment:
This should not be included?

Reply:
This is optional.  Changed to `getProp`.